stuffer/s2n_stuffer_text.c

```diff
@@ -49,6 +49,37 @@ int s2n_stuffer_skip_whitespace(struct s2n_stuffer *s2n_stuffer)
     return skipped;
 }
 
+int s2n_stuffer_read_expected_str(struct s2n_stuffer *stuffer, const char* expected)
+{
+    void* actual = s2n_stuffer_raw_read(stuffer, strlen(expected));
+    notnull_check(actual);
+    if (memcmp(actual, expected, strlen(expected))) {
+        S2N_ERROR(S2N_ERR_STUFFER_NOT_FOUND);
+    }
+    return 0;
+}
+
+int s2n_stuffer_skip_read_until(struct s2n_stuffer *stuffer, const char* target)
+{
+    int len = strlen(target);
+    while (s2n_stuffer_data_available(stuffer) >= len) {
+
+        GUARD(s2n_stuffer_skip_to_char(stuffer, target[0]));
+        char* actual = s2n_stuffer_raw_read(stuffer, len);
+        notnull_check(actual);
+
+        if (strncmp(actual, target, len) == 0){
+            return 0;
+        } else {
+            GUARD(s2n_stuffer_rewind_read(stuffer, len - 1));
```

Comment:
A comment on what this is doing might be helpful. `Reset the stuffer to the next possible starting point`?